src/app/+display/section/section.component.ts

```diff
@@ -16,4 +18,26 @@ export class SectionComponent {
     @Input()
     public document: SolrDocument;
 
+    @Input()
+    public displayViewName: string;
+
+    @Input()
+    public collection: string;
+
+
+    public buildEmbedCode(): string {
+        return '<div class="scholars-embed" data-id="' + this.document.id + '" data-collection="' + this.collection + '" data-displayview="' + this.displayViewName + '" data-sections="' + this.section.name + '"></div>' +
+                '<!-- This javascript only needs to be included once in your HTML -->' +
+                '<script type="text/javascript" src="' + environment.service + '/js/embed/dist/bundle.js" async></script>';
+    }
+
+    public copyToClipBoard(copyElement: any) {
+        copyElement.select();
+        document.execCommand('copy');
+        copyElement.setSelectionRange(0, 0);
+    }
+
+    public isShareable(): boolean {
```

Comment:
This is not an extensible approach to determine which section is to allow sharing embedded snippet or data response. It was discussed that the `DisplayTabSectionView` will have a new property 'shared' alongside 'hidden' to indicate if the section can be shared. Unfortunately, there was no issue created for that.